nix/modules/scripts.nix

```diff
@@ -35,8 +35,15 @@
         set -xeuo pipefail
         trap "cd $PWD" EXIT
 
-        export VERSIONS_DIR="./versions/''${1}"
-        export DEFAULT_VERSIONS_DIR="$(nix flake metadata --no-write-lock-file --json | jq --raw-output '.locks.nodes.versions.locked.path')"
+        # Check the version of the format, otherwise processing the output might fail silently, like using jq again below
+        FLAKES_LOCK_VERSION=$(nix flake metadata --no-write-lock-file --json | jq --raw-output '.locks.version')
+        if [[ "$FLAKES_LOCK_VERSION" != "7" ]]; then
+          echo "Flakes lock version has changed, refusing to update"
+          exit 1
+        fi
+
+        export VERSIONS_DIR="versions/''${1}"
```

Comment:
is the removal of the `./` part of the improvement? might be worth double-checking whether it influences nix interpretation of the path.

Reply:
Used in equality check. It if does use it then Nix is wrong, either should be interpreted as a relative path from the current working directory.